server/src/main/java/org/elasticsearch/action/fieldcaps/TransportFieldCapabilitiesAction.java

```diff
@@ -460,7 +472,13 @@ List<FieldCapabilitiesFailure> build(Set<String> successfulIndices) {
         }
 
         void collect(String index, Exception e) {
-            failuresByIndex.putIfAbsent(index, e);
+            Exception prev = failuresByIndex.putIfAbsent(index, e);
+            if (prev == null
+                && ExceptionsHelper.status(e).getStatus() >= 500
+                && ExceptionsHelper.isNodeOrShardUnavailableTypeException(e) == false) {
+                Supplier<String> messageSupplier = () -> String.format(Locale.ROOT, "Field caps Exception for index: [%s].", index);
+                LOGGER.warn(messageSupplier, e);
+            }
```

Comment:
We don't need both of these logging locations. I've kept both experiments for this draft PR to get feedback.

Reply:
The way I understand execution order after some speed-reading the code, the "collect" gets called earlier than the "build" method above. Also here we would only ever see one exception at once an e.g. wouldn't be able to dedup across different or do counting of similar causes like I suggested above. The other location seems to offer a more global view of all the exceptions that have occured, offering more flexibility in error reporting.
So with my current understanding I would probably opt for the first location. I might miss some advantages of this logging location though.